### PR TITLE
optimize: simplify and groom configuration items

### PR DESCRIPTION
**Pull request by slievrly**
Signed-off-by: slievrly <slievrly@163.com>

### Ⅰ. Describe what this PR did

optimize: simplify and groom configuration items

### CODE REVIEW

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 1]
```diff
@@ -96,17 +101,18 @@
...
      */
-    public static final String CLIENT_LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT = CLIENT_PREFIX + "lock.retry.policy.branch-rollback-on-conflict";
+    public static final String CLIENT_LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT = CLIENT_RM_PREFIX
+        + "lock.retry.policy.branch-rollback-on-conflict";
 
     /**
      * The constant CLIENT_TABLE_META_CHECK_ENABLE.
      */
-    public static final String CLIENT_TABLE_META_CHECK_ENABLE  = CLIENT_PREFIX + "table.meta.check.enable";
+    public static final String CLIENT_TABLE_META_CHECK_ENABLE = CLIENT_PREFIX + "table.meta.check.enable";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: CLIENT_RM_PREFIX

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 2]
```diff
@@ -267,22 +273,22 @@
     /**
      * The constant TRANSACTION_UNDO_LOG_SAVE_DAYS.
      */
-    public static final String TRANSACTION_UNDO_LOG_SAVE_DAYS = TRANSACTION_PREFIX + "undo.log.save.days";
+    public static final String TRANSACTION_UNDO_LOG_SAVE_DAYS = CLIENT_UNDO_PREFIX + "log.save.days";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: On the server side

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 3]
```diff
@@ -267,22 +273,22 @@
...
 
     /**
      * The constant TRANSACTION_UNDO_LOG_DELETE_PERIOD
      */
-    public static final String TRANSACTION_UNDO_LOG_DELETE_PERIOD = TRANSACTION_PREFIX + "undo.log.delete.period";
+    public static final String TRANSACTION_UNDO_LOG_DELETE_PERIOD = CLIENT_UNDO_PREFIX + "log.delete.period";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: On the server side

File: core/src/main/java/io/seata/core/store/db/AbstractDataSourceGenerator.java [hunk 4]
```diff
@@ -43,17 +47,20 @@
...
     protected String getDriverClassName() {
-		String driverClassName = CONFIG.getConfig(ConfigurationKeys.STORE_DB_DRIVER_CLASS_NAME);
-		if (StringUtils.isBlank(driverClassName)) {
-	            throw new StoreException("the {store.db.driver-class-name} can't be empty.");
-	        }
-	    return driverClassName;
-	}
+        String driverClassName = CONFIG.getConfig(ConfigurationKeys.STORE_DB_DRIVER_CLASS_NAME);
+        if (StringUtils.isBlank(driverClassName)) {
+            throw new StoreException("the {store.db.driver-class-name} can't be empty.");
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/store/db/AbstractDataSourceGenerator.java]: recommend ConfigurationKeys.STORE_DB_DRIVER_CLASS_NAME

File: server/src/main/resources/file.conf.example [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+client {
+  rm {
+    async.commit.buffer.limit = 10000
+    lock {
+      retry.internal = 10
+      retry.times = 30
+      retry.policy.branch-rollback-on-conflict = true
+    }
+    report.retry.count = 5
```
[the remark this answers is not in the document]
Comment (slievrly) [reply server/src/main/resources/file.conf.example]: report is the function of rm.

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java (hunk 1 above)
Comment (slievrly) [reply core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: great found, fixed.

File: core/src/main/java/io/seata/core/store/db/AbstractDataSourceGenerator.java (hunk 4 above)
Comment (slievrly) [reply core/src/main/java/io/seata/core/store/db/AbstractDataSourceGenerator.java]: fixed.

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java (hunk 2 above)
Comment (slievrly) [reply core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: fixed.

File: server/src/main/resources/file.conf.example [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    table.meta.check.enable = true
+  }
+  tm {
+    commit.retry.count = 5
+    rollback.retry.count = 5
+  }
+  undo {
```
Comment (zjinlei) [inline server/src/main/resources/file.conf.example]: Is it classified under rm?

**Pull request merged**